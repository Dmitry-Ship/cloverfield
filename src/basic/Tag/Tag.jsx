import React, { PropTypes } from 'react';

import { Link } from 'react-router';

import Icon from '../Icon';

const Tag = ({ tag, id, onRemoveTag }) => {

  const removeTag = () => onRemoveTag(tag, id)

  return (
    <div className="tag">
      <Link className='tag__link' to={`/tags/${tag}`}>
        {tag}
      </Link>
      <Icon className='tag__icon' onClick={removeTag} name='close' size='18px'/>
    </div>
  )
}

export default Tag;

Tag.propTypes = {
  onRemoveTag: PropTypes.func.isRequired,
  tag: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
}
