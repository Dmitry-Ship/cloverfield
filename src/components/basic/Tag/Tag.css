

.tag {
	position: relative;
	display: inline-block;
	margin: 0 5px 5px 0;
	padding: 5px 23px 5px 5px;
	background: rgba(0,0,0,0.3);
	cursor: pointer;
}

.tag:hover .icon {
	opacity: 1;
}

.link {
	color: #fff;
}

.icon {
  position: absolute;
	top: 50%;
  right: 4px;
	color: rgba(255,255,255,0.5);
	opacity: 0;
	transition: all 300ms ease-in-out;
	transform: translateY(-50%);
}

.icon:hover {
	color: #fff;
}
