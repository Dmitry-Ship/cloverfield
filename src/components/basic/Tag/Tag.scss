.tag {
  margin: 0 5px 5px 0;
  background: rgba(black, 0.3);
  cursor: pointer;
  display: inline-block;
  position: relative;
  padding: 5px 30px 5px 5px;
  &:hover .icon {
    opacity: 1;
  }
}

.link {
  color: white;
}

.icon {
  position: absolute;
  top: 50%;
  right: 8px;
  transform: translateY(-50%);
  color: rgba(white, 0.5);
  transition: all 300ms ease-in-out;
  &:hover {
    color: white;   
  }
}
