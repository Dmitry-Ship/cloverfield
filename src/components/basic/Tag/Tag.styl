.tag
  margin 0 5px 5px 0
  background rgba(black, 0.3)
  cursor pointer
  display inline-block
  position relative
  padding 5px 23px 5px 5px
  &:hover
    .tag__icon
      opacity 1

.tag__link
  color white

.tag__icon
  absolute top 50% right 4px
  transform translateY(-50%)
  color rgba(white, 0.5)
  opacity 0
  transition all 300ms ease-in-out
  &:hover
    color white
