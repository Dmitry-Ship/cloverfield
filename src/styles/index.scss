@import '_normalize.scss';
@import '_basic.scss';