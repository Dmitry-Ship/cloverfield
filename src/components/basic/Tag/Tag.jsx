import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import { tag, link, icon } from './Tag.scss';

const Tag = ({ tagText, onDeleteTag }) => (
  <div className={tag}>
    <Link className={link} to={`/tags/${tagText}`}>
      {tagText}
    </Link>
    <span className={icon} onClick={() => onDeleteTag(tagText)}>âœ•</span>
  </div>
);

export default Tag;

Tag.propTypes = {
  onDeleteTag: PropTypes.func.isRequired,
  tagText: PropTypes.string.isRequired,
};
