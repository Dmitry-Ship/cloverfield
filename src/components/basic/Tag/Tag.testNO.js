import React from 'react';
import { shallow } from 'enzyme';

import { Link } from 'react-router';
import Tag from './Tag';


describe('<MyComponent />', () => {
  it('renders correctly without crashing', () => {
    const wrapper = shallow(<Tag />);
    expect(wrapper.find(Link)).to.have.length(1);
  });
});

