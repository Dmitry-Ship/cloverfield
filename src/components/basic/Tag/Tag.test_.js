import React from 'react';
import { shallow, render } from 'enzyme';
// import { describe, it } from 'mocha';
import chai from 'chai';
import { Link } from 'react-router';
import Tag from './Tag';
// import styles from './Tag.scss';
const expect = chai.expect;

describe('<Tag />', () => {
  const tagText = 'lol';
  const onDeleteTag = () => {};
  it('renders correctly without crashing', () => {
    const wrapper = shallow(<Tag tagText={tagText} onDeleteTag={onDeleteTag} />);
    expect(wrapper.find('div')).to.have.length(1);
  });

  it('renders <Link /> component inside', () => {
    const wrapper = shallow(<Tag tagText={tagText} onDeleteTag={onDeleteTag} />);
    expect(wrapper.find('Link')).to.have.length(1);
  });

  it('renders delete icon', () => {
    const wrapper = shallow(<Tag tagText={tagText} onDeleteTag={onDeleteTag} />);
    expect(wrapper.find('span')).to.have.length(1);
  });

  it('renders tagText prop correctly', () => {
    const wrapper = shallow(<Tag tagText={tagText} onDeleteTag={onDeleteTag} />);
    expect(wrapper.find(Link).text()).to.equal(tagText);
  });
  it('deletes tag on click event');
});


