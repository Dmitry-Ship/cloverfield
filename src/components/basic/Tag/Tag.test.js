import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import Tag from './Tag';
import styles from './Tag.scss';

describe('<Tag />', () => {
  const onDeleteTag = jest.fn();
  const tagText = 'hello';

  it('should render correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Tag tagText={tagText} onDeleteTag={onDeleteTag} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onDeleteTag with tagText argument', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Tag tagText={tagText} onDeleteTag={onDeleteTag} />
      </MemoryRouter>
    );

    wrapper.find('span').simulate('click');

    expect(onDeleteTag.mock.calls[0][0]).toBe(tagText);
  });
});
