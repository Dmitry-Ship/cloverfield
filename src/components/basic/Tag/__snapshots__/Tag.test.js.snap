// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render correctly 1`] = `
<div
  className="tag"
>
  <a
    className="link"
    href="/tags/hello"
    onClick={[Function]}
  >
    hello
  </a>
  <span
    className="icon"
    onClick={[Function]}
  >
    âœ•
  </span>
</div>
`;
